scripts/cluster/agent.py

```diff
@@ -423,6 +443,285 @@ def configure():
+    if "service" not in request.json or len(request.json["service"].strip()) == 0:
+        error_msg = {"error": "Empty service provided"}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    if "lines" in request.json:
+        lines = request.json["lines"]
+    # more info on logs could be obtained by using: --output=json-pretty
+    output = subprocess.check_output("journalctl --lines={} --unit=snap.{}.service --no-pager".format(lines, request.json["service"]).split())
+    return output
+
+
+@app.route('/{}/addon/enable'.format(CLUSTER_API), methods=['POST'])
+def enable():
+    """
+    Web call to microk8s.enable <addon>
+    :request_param addon: the name of the addon
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    if "addon" not in request.json or len(request.json["addon"].strip()) == 0:
+        error_msg = {"error": "Empty addon provided"}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    output = subprocess.check_output("{}/microk8s-enable.wrapper {}".format(snap_path, request.json["addon"]).split())
+    return output
+
+
+@app.route('/{}/addon/disable'.format(CLUSTER_API), methods=['POST'])
+def disable():
+    """
+    Web call to microk8s.disable <addon>
+    :request_param addon: the name of the addon
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    if "addon" not in request.json or len(request.json["addon"].strip()) == 0:
+        error_msg = {"error": "Empty addon provided"}
+        return Response(json.dumps(error_msg), mimetype='application/json', status=400)
+    output = subprocess.check_output("{}/microk8s-disable.wrapper {}".format(snap_path, request.json["addon"]).split())
+    return output
+
+
+@app.route('/{}/start'.format(CLUSTER_API), methods=['POST'])
+def start():
+    """
+    Web call for microk8s.start
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    output = subprocess.check_output("{}/microk8s-start.wrapper".format(snap_path).split())
+    return output
+
+
+@app.route('/{}/stop'.format(CLUSTER_API), methods=['POST'])
+def stop():
+    """
+    Web call for microk8s.stop
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    output = subprocess.check_output("{}/microk8s-stop.wrapper".format(snap_path).split())
+    return output
+
+
+@app.route('/{}/overview'.format(CLUSTER_API), methods=['POST'])
+def overview():
+    """
+    Web call to get the microk8s.kubectl get all --all-namespaces
+    """
+    v = callback_token_validation(request)
+    if not v["valid"]: return v["response"]
+    output = subprocess.check_output("{}/microk8s-kubectl.wrapper get all --all-namespaces".format(snap_path).split())
+    return output
+
+
+@app.route('/{}/status'.format(CLUSTER_API), methods=['POST', 'GET'])
```

Comment:
I am not sure about the GET part. Yes it makes sense to be a get but you need to pass the callback token in a secure way.

Reply:
Just my 2 cents, would passing the token through header be acceptable?
The same thing as what most APIs are doing like oauth2.
Wdyt?

Reply:
The way it is implemented now, is allowing GET request with no token passed, if and only if 'enable-extended-api' is true.

Additionally, passing the token through Header or a json body, both ways are supported. Token is never exposed though a query param